src/chttpd/src/chttpd_stats.erl

```diff
@@ -24,29 +24,38 @@
     incr_writes/1,
 
     incr_rows/0,
-    incr_rows/1
+    incr_rows/1,
+
+    update_interval/1
 ]).
 
 
 -record(st, {
     reads = 0,
     writes = 0,
-    rows = 0
+    rows = 0,
+    last_report_ts = 0,
+    interval,
+    request
 }).
 
 
 -define(KEY, chttpd_stats).
+-define(INTERVAL, 60).
 
-
-init() ->
-    put(?KEY, #st{}).
+init(Request) ->
+    Time = erlang:monotonic_time(second),
+    Interval = config:get_integer("chttpd", "stats_reset_interval",
```

Comment:
Perhaps `stats_reporting_interval` might be better since reporting is more important than resetting I think. Reset just happens after we report